recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -159,11 +162,17 @@ def validate(self):
                 f"{', '.join(self._required_boost_components)}"
             )
 
+        if conan_version.major == 1 and self.settings.compiler == "apple-clang" and Version(self.version) >= "1.12.0":
+            # Only fails on apple-clang in this configuration for some reason:
+            # https://github.com/conan-io/conan-center-index/pull/21332#issuecomment-1830766894
+            raise ConanInvalidConfiguration("opentelemetry-cpp >= 1.12.0 does not support Apple Clang on Conan v1")
+
     def build_requirements(self):
-        if self.options.with_otlp:
-            self.tool_requires("protobuf/3.21.9")
-            if self.options.get_safe("with_otlp_grpc"):
-                self.tool_requires("grpc/1.50.1")
+        if self.options.with_otlp or self.options.with_otlp_grpc:
+            self.tool_requires("opentelemetry-proto/1.0.0")
+            self.tool_requires("protobuf/<host_version>")
+            if self.options.with_otlp_grpc:
+                self.tool_requires("grpc/<host_version>")
```

Comment:
```suggestion
        if self.options.with_otlp_grpc:
            self.tool_requires("opentelemetry-proto/1.0.0")
            self.tool_requires("protobuf/<host_version>")
            self.tool_requires("grpc/<host_version>")
```
Please, do not include deprecated options as part of the condition flow.